packages/frontend/src/components/attachment/mediaAttachment.tsx

```diff
@@ -609,44 +610,97 @@ export function WebxdcAttachment({
     tabIndex: rovingTabindex.tabIndex,
   } as const
 
-  if (loadResult.kind === 'loadingError') {
+  useEffect(() => {
+    if (loadResult.kind === 'message') {
+      setIsLoadingWebxdcInfo(true)
+      BackendRemote.rpc
+        .getWebxdcInfo(accountId, messageId)
+        .then(info => {
+          if (info) {
+            setWebxdcInfo(info)
+          } else {
+            setWebxdcInfo(null)
+          }
+        })
+        .catch(error => {
+          log.error(
+            'Failed to load webxdc info for message:',
+            messageId,
+            'accountId:',
+            accountId,
+            'error:',
+            error
+          )
+          setWebxdcInfo(null)
+        })
+        .finally(() => {
+          setIsLoadingWebxdcInfo(false)
+        })
+    }
+  }, [accountId, messageId, loadResult.kind])
+
+  // If the message is not yet loaded by GridGallery, show loading state
+  if (!loadResult || loadResult.kind !== 'message') {
+    return (
+      <div
+        ref={interactiveElRef}
+        className={'media-attachment-webxdc ' + rovingTabindex.className}
+        {...rovingTabindexProps}
+      >
+        <img
+          className='icon'
+          src={runtime.getWebxdcIconURL(selectedAccountId(), messageId)}
+        />
+        <div className='text-part'>
+          <div className='name'>Loading...</div>
+          <div className='summary'></div>
+        </div>
+      </div>
```

Comment:
It would be nice not to copy-paste this.

Reply:
I agree but that should go into another PR - this PR is already pretty overloaded anyway